library/Booster/Pattern/ApplyEquations.hs

```diff
@@ -704,17 +708,17 @@ simplifyConstraint' = \case
                     evalBool t >>= prune
         | otherwise ->
             evalBool t >>= prune
-    EqualsTerm TrueBool t ->
-        -- although "true" is usually 2nd
-        simplifyConstraint' (EqualsTerm t TrueBool)
+    EqualsTerm t TrueBool ->
+        -- normalise to 'true' in first argument (like 'kore-rpc')
+        simplifyConstraint' (EqualsTerm TrueBool t)
     other ->
-        pure other -- should not occur, predicates should be '_ ==Bool true'
+        pure other -- should not occur, predicates should be 'true \equals _'
```

Comment:
This changes the preferred order of the equals-predicates `P  \equals true` , to `true \equals P` which aligns with the `kore-rpc` behaviour, and required adapting test expectations.
Because we don't run a 0-step execute request after the internal simplification any more, we are now getting booster's preference in more cases.